core/src/apps/management/reset_device.py

```diff
@@ -123,18 +126,49 @@
     await layout.slip39_show_and_confirm_shares(ctx, mnemonics)
 
 
+async def backup_group_slip39_wallet(ctx: wire.Context, secret: bytes) -> None:
+    # get number of groups
+    await layout.slip39_group_show_checklist_set_groups(ctx)
+    groups_count = await layout.slip39_prompt_number_of_groups(ctx)
+
+    # get group threshold
+    await layout.slip39_group_show_checklist_set_group_threshold(ctx, groups_count)
+    group_threshold = await layout.slip39_prompt_group_threshold(ctx, groups_count)
+
+    # get shares and thresholds
+    await layout.slip39_group_show_checklist_set_shares(
+        ctx, groups_count, group_threshold
+    )
+    groups = []
+    for i in range(groups_count):
+        share_count = await layout.slip39_prompt_number_of_shares(ctx, i)
+        share_threshold = await layout.slip39_prompt_threshold(ctx, share_count, i)
+        groups.append((share_threshold, share_count))
+
+    # generate the mnemonics
+    mnemonics = slip39.generate_mnemonics_from_data(
+        encrypted_master_secret=secret,
+        identifier=storage.device.get_slip39_identifier(),
+        group_threshold=group_threshold,
+        groups=groups,
+    )
+
+    # show and confirm individual shares
+    await layout.slip39_group_show_and_confirm_shares(ctx, mnemonics)
+
+
 async def backup_bip39_wallet(ctx: wire.Context, secret: bytes) -> None:
     mnemonic = bip39.from_data(secret)
     await layout.bip39_show_and_confirm_mnemonic(ctx, mnemonic)
 
 
 def _validate_reset_device(msg: ResetDevice) -> None:
     msg.backup_type = msg.backup_type or _DEFAULT_BACKUP_TYPE
-    if msg.backup_type not in (
-        ResetDeviceBackupType.Bip39,
-        ResetDeviceBackupType.Slip39_Single_Group,
-    ):
-        raise wire.ProcessError("Backup type not implemented.")
+    # if msg.backup_type not in (
+    #     ResetDeviceBackupType.Bip39,
+    #     ResetDeviceBackupType.Slip39_Single_Group,
+    # ):
+    #     raise wire.ProcessError("Backup type not implemented.")
```

Comment:
Just add `Slip39_Multiple_Groups` and keep it :).

Reply:
done, not sure why I commented it out in the first place :)